hnn_core/network.py

```diff
@@ -488,3 +479,66 @@ def plot_spikes(self, ax=None, show=True):
         if show:
             plt.show()
         return ax.get_figure()
+
+    def build_in_neuron(self):
+        """This function must be called before Network can be used for sims"""
+
+        from neuron import h
+        from .parallel import create_parallel_context
+        from .parallel import clear_last_network_objects
+
+        # make sure ParallelContext has been created (needed for joblibs)
+        create_parallel_context()
+
+        clear_last_network_objects(self)
+
+        self._gid_assign()
+        # Create a h.Vector() with size 1xself.N_t, zero'd
+        self.current = {
+            'L5Pyr_soma': h.Vector(self.N_t, 0),
+            'L2Pyr_soma': h.Vector(self.N_t, 0),
+        }
+        self._create_all_src()
+        self.state_init()
+        self._parnet_connect()
+        # set to record spikes
+        self.spiketimes = h.Vector()
+        self.spikegids = h.Vector()
+        self._record_spikes()
+        # position cells in 2D grid
+        self.move_cells_to_pos()
+
+    def _clear_neuron_objects(self):
+        """
+        Clear up NEURON internal gid information.
+
+        Note: This function must be called from the context of the
+        Network instance that ran build_in_neuron. This is a bug or
+        peculiarity of NEURON. If this function is called from a different
+        context, then the next simulation will run very slow because nrniv
+        workers are still going for the old simulation. If pc.gid_clear is
+        called from the right context, then those workers can exit.
```

Comment:
@blakecaldwell I stumbled upon this issue myself. I think we should raise a ticket on Neuron github so it is documented at least if it cannot be fixed. cc @ramcdougal

Reply:
@jasmainak okay. on my list

Reply:
@jasmainak Do you have any new ideas after working with the other NEURON issue with VecStim?